crates/sc-proof-of-time/src/node_client.rs

```diff
@@ -58,10 +59,22 @@ where
             Arc::new(|sender, proof| {
                 self.handle_gossip_message(sender, proof);
             });
-        self.gossip
-            .process_incoming_messages(handle_gossip_message)
-            .await;
-        error!("Gossip engine has terminated");
+        let mut block_import = self.client.import_notification_stream();
+        loop {
+            futures::select! {
+                incoming_block = block_import.next().fuse() => {
+                    if let Some(incoming_block) = incoming_block {
+                        self.handle_block_import(incoming_block);
+                    }
+                }
```

Comment:
Here you assume that block at a specific height will be imported only once. This is not the case, it can be imported more than once in case of forks. Moreover, you're subscribing to notification of the canonical chain, so if reorg does happen, you will not catch it and PoT will use incorrect block hash.